### PR TITLE
err.description is deprecated

### PR DESCRIPTION
**Pull request by samuela**
Fixes compile warnings in juniper_hyper:
```
❯ cargo build
warning: use of deprecated item 'std::error::Error::description': use the Display impl or to_string()
   --> juniper_hyper/src/lib.rs:376:60
    |
376 |             GraphQLRequestError::BodyHyper(ref err) => err.description(),
    |                                                            ^^^^^^^^^^^
    |
    = note: `#[warn(deprecated)]` on by default

warning: use of deprecated item 'std::error::Error::description': use the Display impl or to_string()
   --> juniper_hyper/src/lib.rs:377:59
    |
377 |             GraphQLRequestError::BodyUtf8(ref err) => err.description(),
    |                                                           ^^^^^^^^^^^

warning: use of deprecated item 'std::error::Error::description': use the Display impl or to_string()
   --> juniper_hyper/src/lib.rs:378:64
    |
378 |             GraphQLRequestError::BodyJSONError(ref err) => err.description(),
    |                                                                ^^^^^^^^^^^

warning: use of deprecated item 'std::error::Error::description': use the Display impl or to_string()
   --> juniper_hyper/src/lib.rs:379:60
    |
379 |             GraphQLRequestError::Variables(ref err) => err.description(),
    |                                                            ^^^^^^^^^^^

    Finished dev [unoptimized + debuginfo] target(s) in 0.43s
```

### CODE REVIEW

File: juniper_hyper/src/lib.rs [hunk 1]
```diff
@@ -371,16 +371,6 @@ impl fmt::Display for GraphQLRequestError {
...
-        match *self {
-            GraphQLRequestError::BodyHyper(ref err) => err.description(),
-            GraphQLRequestError::BodyUtf8(ref err) => err.description(),
-            GraphQLRequestError::BodyJSONError(ref err) => err.description(),
-            GraphQLRequestError::Variables(ref err) => err.description(),
-            GraphQLRequestError::Invalid(ref err) => err,
-        }
-    }
-
     fn cause(&self) -> Option<&dyn Error> {
```
Comment (andy128k) [inline juniper_hyper/src/lib.rs]: `cause` is also deprecated and replaced by `source`.

Comment (samuela) [reply juniper_hyper/src/lib.rs]: Interesting... do you know why this warning isn't also being surfaced? And should I add that change to this PR or is it out of scope?

Comment (andy128k) [reply juniper_hyper/src/lib.rs]: Warning is triggered by usage (invocation) of deprecated method, not by its declaration.

Comment (samuela) [reply juniper_hyper/src/lib.rs]: Ok, fixed `cause()` in f840576

**Comment by LegNeato** [comment]
Thanks for the PR! 🥂

**Pull request merged**